src/common/types/data_chunk.cpp

```diff
@@ -552,7 +561,12 @@ void SetArrowChild(DuckDBArrowArrayChildHolder &child_holder, const LogicalType
 			if (!mask.RowIsValid(row_idx)) {
 				continue;
 			}
-			auto &str = string_t_ptr[row_idx];
+			string_t str;
+			if (type.id() == LogicalTypeId::UUID) {
+				str = UUID::ToString(uint64_t_ptr[row_idx]);
```

Comment:
This is undefined behavior, `UUID::ToString` returns a `std::string`, `string_t` will then get a pointer to that string, after which the string gets deleted. This will likely crash.